### ISSUE TITLE
Form main functionalities like get data may be exported too

### ISSUE THREAD

**Issue by italomlp** [Rocketseat/unform#76]
**What would you like to be added**:
Some functionalities may be exported from the form itself, like getting data outside the onSubmit, reset form outside onSubmit too, and so on.

**Why is this needed**:
I think that it is a good feature to introduce. Some developers want to control the form by its hand. So with this functionalities, they can control the form without the need to be in the on submit function. 

E.g.: get data of the form without submitting, or clear data manually.

**Is your enhancement request related to a problem? Please describe.**
No.

**Additional context**

I was working on #73, and face this. I already have the code in mind (and already made a working getData function). What do you guys think about this?

**Comment by diego3g** [comment]
I think it would be great. What do you have in mind, maybe using Render Props pattern like this?
```js
<Form>
  { ({ getValue, setValue }) => (
    // form fields
  )}
</Form>
```

Or maybe even better finding a way of using React Hooks to do this job, i didn't think about it but maybe something like:
```js
const [Form, { getValue, setValue }] = useForm();

return <Form />;
```

**Comment by PejmanNik** [comment]
I also need setError for setting API error. #88. but I think for that case add this method to submit helper function is just fine.

and about the setValue. I'm not sure my proposal is good or even isn't an anti-pattern. but:
we can pass a helper function object to Form component. then component fills it with self helper option. so parent component just can call that object function easily. like what we do with refs.

**Comment by diego3g** [comment]
Unform released version 2.0 🎉

Now you can access a lot of form functionalities using the form reference, check [this guide](https://unform.dev/recipes/accessing-form-ref).
